compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/pager/Pager.kt

```diff
@@ -895,7 +895,10 @@ private class DefaultPagerNestedScrollConnection(
         available: Offset,
         source: NestedScrollSource
     ): Offset {
-        if (source == NestedScrollSource.Fling && available != Offset.Zero) {
+        if (source == NestedScrollSource.Fling &&
+            (orientation == Orientation.Horizontal && available.x != 0f) ||
+            (orientation == Orientation.Vertical && available.y != 0f)
+        ) {
```

Comment:
Let's copy approach from Scrollable:

Add:
```kt
    private fun Offset.toFloat(): Float =
        if (orientation == Orientation.Horizontal) this.x else this.y
```

And replace

```suggestion
        if (source == NestedScrollSource.Fling && available.toFloat() != 0f) {
```

Reply:
Dure, done